src/main.cpp

```diff
@@ -5530,6 +5530,25 @@ bool ProcessMessages(CNode* pfrom)
     return fOk;
 }
 
+class CompareInvMempoolOrder
+{
+    CTxMemPool *mp;
+public:
+    CompareInvMempoolOrder(CTxMemPool *mempool)
+    {
+        mp = mempool;
+    }
+
+    bool operator()(const CInv &a, const CInv &b)
+    {
+        if (a.type != MSG_TX) {
```

Comment:
Have you verified that using this comparator does not change the order of blocks?

Alternatively, if both entries are blocks, perhaps they should be sorted by height?